examples/step-92/step-92.cc

```diff
@@ -0,0 +1,1091 @@
+
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+
+
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/data_out_faces.h>
+#include <deal.II/numerics/error_estimator.h>
+
+
+// Utilities::System namespace will be used to query things like the
+// number of processors associated with the current MPI universe, or the
+// number within this universe the processor this job runs on is:
+#include <deal.II/base/utilities.h>
+
+// class ConditionOStream allows to write
+// code that would output things to a stream (such as <code>std::cout</code>)
+// on every processor but throws the text away on all but one of them.
+#include <deal.II/base/conditional_ostream.h>
+
+// Indicate which elements a particular
+// processor owns or need to know of. This is the realm of the IndexSet class:
+// if there are a total of $N$ cells, degrees of freedom, or vector elements,
+// associated with (non-negative) integral indices $[0,N)$, then both the set
+// of elements the current processor owns as well as the (possibly larger) set
+// of indices it needs to know about are subsets of the set $[0,N)$. IndexSet
+// is a class that stores subsets of this set in an efficient format:
+#include <deal.II/base/index_set.h>
+
+// The next header file is necessary for a single function,
+// SparsityTools::distribute_sparsity_pattern:
+#include <deal.II/lac/sparsity_tools.h>
+
+// parallel::distributed::Triangulation provide meshes distributed
+// across a potentially very large number of processors, while the second
+// provides the namespace parallel::distributed::GridRefinement that offers
+// functions that can adaptively refine such distributed meshes:
+#ifdef USE_FULLY_DISTRIBUTED_TRIA
+#  include <deal.II/distributed/fully_distributed_tria.h>
+#else
+#  include <deal.II/distributed/tria.h>
+#endif
+#include <deal.II/distributed/grid_refinement.h>
+
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+
+
+// @sect3{Class Template Declarations}
+//
+// Declaration of all classes with their data structures and methods
+
+namespace Step92
+{
+  template <typename T>
+  T sum(const T &, const T &);
+
+  template <typename T>
+  double sum(const double &a, const double &b)
+  {
+    return a + b;
+  };
+
+  using namespace dealii;
+  using namespace std::complex_literals;
+
+  // @sect4{Parameters Class}
+
+  // The Parameters class inherits ParameterAcceptor, and instantiates all the
+  // coefficients in the variational equations.
+  // These coefficients are passed through ParameterAcceptor and are editable
+  // through a .prm file.
+
+
+  template <int dim>
+  class ProblemParameters : public ParameterAcceptor
+  {
+  public:
+    ProblemParameters();
+    ~ProblemParameters();
```

Comment:
```suggestion
    ~ProblemParameters() = default;
```
and you can then delete the implementation that follows later.